From fe9b7393aff3325b40edc86e9d44cc9c6e75704e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Macholda?= <tomas.macholda@nic.cz>
Date: Tue, 7 Jul 2026 17:47:52 +0200
Subject: [PATCH] mvebu: turris-mox: fix USB2 detection
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix Turris MOX not detecting USB2 devices by explicitly adding all USB3
PHYs and PHY names to Turris MOX device-tree.

Signed-off-by: Tomáš Macholda <tomas.macholda@nic.cz>
---
 ...armada-3720-turris-mox-fix-usb3-phys.patch | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)
 create mode 100644 target/linux/mvebu/patches-6.6/101-arm64-dts-armada-3720-turris-mox-fix-usb3-phys.patch

diff --git a/target/linux/mvebu/patches-6.6/101-arm64-dts-armada-3720-turris-mox-fix-usb3-phys.patch b/target/linux/mvebu/patches-6.6/101-arm64-dts-armada-3720-turris-mox-fix-usb3-phys.patch
new file mode 100644
index 0000000000..8bbc9a2e17
--- /dev/null
+++ b/target/linux/mvebu/patches-6.6/101-arm64-dts-armada-3720-turris-mox-fix-usb3-phys.patch
@@ -0,0 +1,42 @@
+From 57b95de489769259fee99a4b70f10cedc6f24641 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Macholda?= <tomas.macholda@nic.cz>
+Date: Tue, 7 Jul 2026 13:48:03 +0200
+Subject: [PATCH mvebu-dt64] arm64: dts: turris-mox: fix usb3 phys
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+After commit 00e6d608fe80b0f6 ("arm64: dts: marvell: armada-37xx: swap
+PHYs' order in USB3 controller node") swapped USB3 PHY order, USB
+initialization breaks on Turris MOX.
+
+This regression was exposed by commit 91ddf6f722084383 ("phy: marvell:
+mvebu-a3700-utmi: fix incorrect USB2_PHY_CTRL register access") which
+made USB2 devices not work at all.
+
+Fix the issue by explicitly adding all USB3 PHYs and PHY names to
+Turris MOX device-tree.
+
+Fixes: 7109d817db2e ("arm64: dts: marvell: add DTS for Turris Mox")
+Signed-off-by: Tomáš Macholda <tomas.macholda@nic.cz>
+---
+ arch/arm64/boot/dts/marvell/armada-3720-turris-mox.dts | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/arch/arm64/boot/dts/marvell/armada-3720-turris-mox.dts b/arch/arm64/boot/dts/marvell/armada-3720-turris-mox.dts
+index f4d73c8b1a6d..bd00e7df87c6 100644
+--- a/arch/arm64/boot/dts/marvell/armada-3720-turris-mox.dts
++++ b/arch/arm64/boot/dts/marvell/armada-3720-turris-mox.dts
+@@ -290,7 +290,8 @@ connector {
+ 
+ &usb3 {
+ 	status = "okay";
+-	phys = <&comphy2 0>;
++	phys = <&usb2_utmi_otg_phy>, <&comphy2 0>;
++	phy-names = "usb2-phy", "usb3-phy";
+ };
+ 
+ &mdio {
+-- 
+2.54.0
+
-- 
2.53.0

